src/main/java/com/google/sps/servlets/getVideosServlet.java

```diff
@@ -79,46 +80,64 @@ public void doPut (HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     GetServletsUtility.deleteResultsOfQueryFromDatastore(query, datastore,KIND);
 
-    // Get string from api.
+    // Map to store strings and catagory types
+    HashMap<String, String> videoMap = new HashMap<String, String>();
+
+    // Get yoga video string from api.
     String youtubeapiKey = getSecretKey.accessSecretVersion("298755462", "youtube_api_key", "2"); // Get hidden api key from gcloud secret manager.
-    URL url = new URL(baseURL + youtubeapiKey);
-    StringBuilder strBuf = new StringBuilder();  
-    String videos = getStringFromAPI.getStringFromAPIMethod(url, strBuf, null, null, logger, request, response);
-    if (videos.equals("end")) return; // Returns if exception caught.
-
-    // Put data from api string into database.
-    JSONObject obj = new JSONObject(videos);
-    JSONArray videoData = obj.getJSONArray("items");
-    int numVideos = videoData.length();
-    
-    String contentDetailsURL = "https://www.googleapis.com/youtube/v3/videos?part=contentDetails&id=";
-    String videoLinkBaseURL = "https://www.youtube.com/watch?v=";
-
-    for (int i = 0; i < numVideos; ++i) {
-      JSONObject currentVideo = videoData.getJSONObject(i);
-      Entity videoEntity = new Entity(KIND);
-
-      // Get video details from api.
-      String videoId = currentVideo.getJSONObject("id").getString("videoId"); // Get videoId.
-      URL getContentDetailsURL = new URL(contentDetailsURL + videoId + "&key=" + youtubeapiKey);
-      StringBuilder videoStrBuf = new StringBuilder();  
-      String videoDetails = getStringFromAPI.getStringFromAPIMethod(getContentDetailsURL, videoStrBuf, null, null, logger, request, response);
-      if (videos.equals("end")) return; // Returns if exception caught.
-
-      // Get duration from parsed string from api.
-      JSONObject videoDetailObj = new JSONObject(videoDetails);
-      String duration = videoDetailObj.getJSONArray("items").getJSONObject(0).getJSONObject("contentDetails").getString("duration");
-
-      // Convert duration string into seconds.
-      long durationInSeconds = Duration.parse(duration).getSeconds();
+    URL yogaUrl = new URL(baseURL + "yoga" + "&type=video&key=" + youtubeapiKey); 
+    String yogaVideos = getStringFromAPI.getStringFromAPIMethod(yogaUrl, null, null, logger, request, response);
+    if (yogaVideos.equals("end")) return; // Returns if exception caught.
```

Comment:
A comment is printed telling the user what is happening in the getStringFromAPI method, you can see it in the getStringFromAPI.java class where it says "response.getWriter().println", also here's a screenshot of the code: [getStringFromAPI.java file](https://screenshot.googleplex.com/a02a21cb-4490-45bd-8471-98627ea9801a.png)

Reply:
Ok I see.

Reply:
I think it's fine then.

Reply:
But instead of returning "end" when something didn't go right, I think returning null would be better. It's the conventional way of letting the caller know that something went wrong.

Reply:
Also make sure to convey this in the method's java doc

Reply:
Amazing -- thanks! I'll add that now.